packages/api-codec/src/codec/Length.ts

```diff
@@ -6,83 +6,39 @@ import { AnyNumber } from '../types';
 
 import BN from 'bn.js';
 import bnToBn from '@polkadot/util/bn/toBn';
-import bnToU8a from '@polkadot/util/bn/toU8a';
-import u8aConcat from '@polkadot/util/u8a/concat';
-import u8aToBn from '@polkadot/util/u8a/toBn';
 import u8aToHex from '@polkadot/util/u8a/toHex';
 
 import Base from './Base';
+import Compact from './Compact';
 import UInt from './UInt';
 
-const MAX_U8 = new BN(2).pow(new BN(8 - 2)).subn(1);
-const MAX_U16 = new BN(2).pow(new BN(16 - 2)).subn(1);
-const MAX_U32 = new BN(2).pow(new BN(32 - 2)).subn(1);
+type BitLength = 32;
 
-// A new compact length-encoding algorithm. It performs the same function as Length, however
-// differs in that it uses a variable number of bytes to do the actual encoding. From the Rust
-// implementation for compact encoding
-//
-//     0b00 00 00 00 / 00 00 00 00 / 00 00 00 00 / 00 00 00 00
-// (0 ... 2**6 - 1)    (u8)
-//     xx xx xx 00
-// (2**6 ... 2**14 - 1)  (u8, u16)  low LH high
-//     yL yL yL 01 / yH yH yH yL
-// (2**14 ... 2**30 - 1)  (u16, u32)  low LMMH high
-//     zL zL zL 10 / zM zM zM zL / zM zM zM zM / zH zH zH zM
-// (2**30 ... 2**536 - 1)  (u32, u64, u128, U256, U512, U520) straight LE-encoded
-//     nn nn nn 11 [ / zz zz zz zz ]{4 + n}
-//
-// Note: we use *LOW BITS* of the LSB in LE encoding to encode the 2 bit key.
+// A Length-prefix that can be added to any variable length stream of bytes, e.g. String,
+// Vectors and Bytes
 //
 // FIXME Not crazy about the use of Length. If we look at Vector, Bytes & String,
 // the implementations are basically the same. Vector is slightly different since
 // it iterates over the length. We need a cleaner way where we have less copied
 // code and a more generic implementation around the use of Length. Looking at
 // Array or Struct, the same type of wrapper would be useful here.
 export default class Length extends Base<BN> {
-  constructor (value: AnyNumber = new BN(0)) {
+  private _bitLength: BitLength;
+
+  constructor (value: AnyNumber = new BN(0), bitLength: BitLength = 32) {
```

Comment:
Bit out of scope for this PR, so not changing things here... but since we have pulled the encoding logic into Compact, we would actually get rid of this use - rather, just call Compact.encode() with the length directly in the variable length uses instead of replying on Length. (Which as per the FIXME has always been a bit of a funny step-child)